### ISSUE TITLE
Bug? Page autocompletion aborts after hyphen (special character)

### ISSUE THREAD

**Issue by grg8** [cosmocode/dokuwiki-plugin-struct#579]
Hi!

First of all, Dokuwiki is great! I hope that this issue has not been mentioned, yet, or i missed an information, and it is explained halfway decent.

In our school wiki there are a bunch of devices documented via most recent stable dokuwiki with struct plugin where the pagenames are corresponding to their names. For example in case of PCs this name is also the hostname, e.g. the full path is `anlage:pc:stock_1_inf1:inf1-pc-01`.

According to https://www.dokuwiki.org/pagename:

```
Allowed characters are letters, digits and, within names 1), the “special characters” ., - and _."

1) i.e. you cannot start or end names with them, resp. they will be eliminated at start or end of names
```

When I enter a hyphen (`-`) in a lookup table inside type page text entry then one more character is accepted by autocompletion but not filtered correctly. After the second character the autocompletion list aborts and shows no result anymore. This happens to all "allowed special characters", thus `.`, `-` and `_`. 

Works:
![struct_01](https://user-images.githubusercontent.com/83758519/123089630-7bc12280-d427-11eb-862f-e811b9f7505e.png)

Aborts:
![struct_02](https://user-images.githubusercontent.com/83758519/123089665-867bb780-d427-11eb-9a54-9029dd17f3c6.png)

My schema for this field:
![struct_03](https://user-images.githubusercontent.com/83758519/123091009-22f28980-d429-11eb-8a02-1738d81159fb.png)

Anyways, thank you for your great work!

**Comment by annda** [comment]
@grg8 Thank you for the excellently detailed report.

TLDR Do not use namespace filter if you have special characters in your page ids.

The problem lies in how the DokuWiki core handles search requests. Page ids are internally parsed differently (in `_ft_pageLookup()`) when a namespace filter is used. Without the filter in autocomplete settings the search with special characters will work fine. With the filter, the page id is treated as a regular query string and all special characters are stripped and words are split. In your case, the one-character word "n" is ignored, so the search for "langer" works. In the longer search in your example the two words "langer" and "na" are internally transformed into "langer_na", which do not match the "longer-na" search pattern.

@splitbrain Should this be addressed in the DokuWiki core's search?

**Comment by splitbrain** [comment]
interesting. I think this would be addressed when implementing #593 using variant 2.

Additionally addressing the issue in core might be a good idea anyway.
